File: cpp/gastos4Semanas.cpp

#include <iostream>
#include <locale.h>
#include <string>
#include <vector>
#include <sstream>

using namespace std;

int main(){
	setlocale(LC_ALL, "");
	
	int total=0, maior=-1, menor = 1000;

	string semana;

	while(getline(cin, semana)){
		istringstream stream(semana);
		for(int dia; stream >> dia; ){
			total += dia;
			if(dia > maior){
				maior = dia;
				if(dia < menor){ menor = dia; }
			} else if (dia < menor){ menor = dia; }
		}
		cout << total << " " << menor << " " << maior << endl;
		total = 0;
		maior = -1;
		menor = 1000;
	
	}
	return 0;
}

File: java/gastos4Semanas.java
/*
 * Laboratório de Programação 2 - Lab 1
 *
 * @author Franklin Regis de Oliveira - 119210030
 */

import java.util.Scanner;

public class gastos4Semanas{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		int total = 0, menor = 10000000, maior = -10000000;
		Integer dia = 0;
		for(int i=0; i<4;i++){
			String[] semana = sc.nextLine().split(" ");
			for(int j=0; j<7; j++){
				dia  = Integer.parseInt(semana[j]);
				total += dia;
				if(dia > maior){maior = dia;}
				if(dia < menor){menor = dia;}
			}
			System.out.println(total + " " + menor + " " + maior);
			total = 0; maior=-100000000; menor = 10000000;
		}
	}
}
